### DISCUSSION TITLE
Imgui Vulkan + SDL2 completely fails in Release but works perfectly in Debug?

### DISCUSSION THREAD

**Discussion by emctague** [ocornut/imgui#4123] [Old/Archived discussions]
Putting this in Discussion because I think there's about a 60% chance it's something dumb on my part.

I'm using IMGUI with Vulkan and SDL2. It works perfectly, no errors, in Debug builds, but when producing a Release build on windows (edit: and macOS too), the default font literally wasn't present. This forced me to add into my initialization, between `CreateContext` and the Impl Init codes:

```cpp
    io.Fonts->AddFontDefault();
    io.Fonts->Build();
```

After this, however, IMGUI Vulkan apparently still isn't in the mood to draw correctly:

```
VUID-vkCmdDrawIndexed-None-02699(ERROR / SPEC): msgNum: -1539028524 - Validation Error: [ VUID-vkCmdDrawIndexed-None-026
99 ] Object 0: handle = 0x57b9520000000037, type = VK_OBJECT_TYPE_DESCRIPTOR_SET; | MessageID = 0xa44449d4 | VkDescripto
rSet 0x57b9520000000037[] encountered the following validation error at vkCmdDrawIndexed() time: Descriptor in binding #
0 index 0 is being used in draw but has never been updated via vkUpdateDescriptorSets() or a similar call. The Vulkan sp
ec states: Descriptors in each bound descriptor set, specified via vkCmdBindDescriptorSets, must be valid if they are st
atically used by the VkPipeline bound to the pipeline bind point used by this command (https://vulkan.lunarg.com/doc/vie
w/1.2.170.0/windows/1.2-extensions/vkspec.html#VUID-vkCmdDrawIndexed-None-02699)
    Objects: 1
        [0] 0x57b9520000000037, type: 23, name: NULL
```
I have validated that these errors originate from ImGui and not from my own rendering.

Any idea why the ImGui_impl_vulkan doesn't properly handle descriptor sets in Release, but works perfectly in Debug?

([more code from my usage of imgui_impl_vulkan, including the relevant CMake entries](https://gist.github.com/emctague/3b6ac5740534fb5e7d01b234bae0bb3f))

**Comment by emctague** [comment]
...for anyone reading this, all I can say is... I put `assert()` around some critical calls!

:D end me

**Comment by YusufSuleman** [comment]
Try to make this an Issue and reveal the whole code and some IDE Screenshots.
